src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -197,16 +213,23 @@ public void Advance(int bytes)
                     return;
                 }
 
-                if (_autoChunk)
+                if (!_startCalled)
                 {
-                    if (bytes < 0)
+                    if (bytes >= 0)
                     {
-                        throw new ArgumentOutOfRangeException(nameof(bytes));
```

Comment:
Do you still throw if bytes is less than zero?

Reply:
Yeah, in HttpProtocol now.